docs/knowledgebase/advanced/account-info.md

```diff
@@ -0,0 +1,149 @@
+  /// The additional data that belongs to this account. Used to store the balance(s) in a lot of
+  /// chains.
+  pub data: AccountData,
+}
+```
+
+Every account is an `AccountInfo` consisting of an `nonce` indicating number of transactions the
+account has sent, three reference counters, and an `AccountData` structure which contains some
+balances ([further explained below](#accountdata-structure)).
+
+Let's look deeper into the three reference counters, `consumers`, `providers`, and `sufficients`.
+The reasons these counters exist because Substrate has a mechanism of slashing accounts (e.g. one
+reason could be an account balance is below the existential deposit). If any of these reference
+counters are greater than zero, it is telling Substrate to not slash the account. For:
+
+- **`consumers` reference counter** indicates if there are any other modules depend on this account.
+An example of using this counter is in `Session` pallet when an account setting its session key(s)
+prior of becoming a validator [[1]](#ref-session-set-keys). It is also related to the `provider`
+counter, see below.
+
+- **`providers` reference counter** indicates if there are any other modules allowing this account
+to exist. Currently this counter  is incremented when a new account is created with more than the
+existential deposit to indicate this account could be used to receive consumer reference counters
+[[2]](#ref-system-created). `providers` counter is always greater than or equal to `consumers`
+counter.
+
+  Both `providers` and `consumers` tell Substrate pallets not to store data about that account until
+  it is active (i.e. `providers` > 0), and not to remove an account until data about the account are
+  removed in all pallets (i.e. `consumers` == 0). This is to keep users accountable for their data
+  stored on chain. If users want to remove their accounts and get back the exisitential deposit,
+  they need to remove the dependencies from those on-chain pallets, e.g. clearing data stored
+  on-chain for those pallets, which decrement `consumers` counter.
+
+- **`sufficients` reference counter** indicates if there are any reasons this account is
+self-sufficient to exist by itself. An example of using this counter is in `Assets` pallet when an
+account has sufficient amount of certain assets but not owning any native account balance
+[[3]](#ref-assets-new-account).
+
+Runtime developers can update these counters via `inc_consumers()`, `dec_consumers()`,
+`inc_providers()`, `dec_providers()`, `inc_sufficients()`, `dec_sufficients()` exposed by
+`frame-system`. Each increment call of a certain counter should accompanied by a corresponding
+decrement call of the counter in an account life cycle, else it is a design bug.
+
+## `AccountData` Structure
+
+The last piece of data in `AccountInfo` is `AccountData`. It is defined in `frame-system` to be
+bound by `Member + FullCodec + Clone + Default` traits. Its actual implementation is defined in
```

Comment:
> bound by `Member + FullCodec + Clone + Default`

You are just stating facts, but this is not at all helpful to the documentation.